### ISSUE TITLE
RGB, lux and color temperature

### ISSUE THREAD

**Issue by aPonza** [ros/common_msgs#133]
I will be using sensor [TCS34725](https://www.adafruit.com/product/1334) and from it's [python library](https://github.com/adafruit/Adafruit_CircuitPython_TCS34725) I can see it provides the user with:
- RGB color data (0-255 values);
- color temperature (in K);
- illuminance (in lux).

The sensor only reads RGBC data, the last two are calculated values, so nothing I can't do on my own from a published RGB-type message. However, I have found that the only really officially supported message regarding color sensor data is the [illuminance message](http://docs.ros.org/kinetic/api/sensor_msgs/html/msg/Illuminance.html), as there is an [RGBA color message](http://docs.ros.org/kinetic/api/std_msgs/html/msg/ColorRGBA.html) but it seems to be more geared towards a different kind of usage (because of the alpha).

I have previously worked a bit with colors and know about the multitude of standards that exist, so that might be a reason why this hasn't been implemented yet. Regardless, if we stay within the robotics applications, an RGB/sRGB definition could be enough, and would set a standard way to talk about RGB/sRGB colors.

Is this worth talking about as a standard or should I use my own message for my own use case?

**Comment by aPonza** [comment]
It seems like #5 and [this](http://wiki.ros.org/sensor_msgs/Reviews/2012-11-16-Low-Cost-Sensor-Roundup_API_Review) would have been related.

**Comment by tfoote** [comment]
Yeah, there hasn't been many sensors providing single point RGB data so I don't think that there's been much demand for a standard message for it. In general we want to see a message being used successfully before accepting it into common_msgs. So I'd suggest that you prototype your own message and once it's working successfully you propose to add it back here for it to be reviewed.

Note that I'd suggest not trying to include the calculated values, only the raw sensor data. And maybe you could publish the Illuminance on a separate topic using the exisiting message in case other users want to use it.

**Comment by aPonza** [comment]
Ok, sure, it makes sense! And I'm definitely agreeing on calculated values. I'll update this once the sensors arrive and I have something.

**Comment by sloretz** [comment]
ROS 1 is end-of-life (EOL) as of today, May 31st 2025. I am archiving this repository because:

* it only supports ROS 1
* it isn't needed anymore in ROS 2

[If you still rely on ROS 1, read this page to learn about your options.](https://bit.ly/NoeticEOL)
